sky/provision/kubernetes/utils.py

```diff
@@ -453,17 +524,12 @@ def get_kubernetes_nodes(context: Optional[str] = None) -> List[Any]:
     if context is None:
         context = get_current_kube_config_context_name()
 
-    try:
-        nodes = kubernetes.core_api(context).list_node(
-            _request_timeout=kubernetes.API_TIMEOUT).items
-    except kubernetes.max_retry_error():
-        raise exceptions.ResourcesUnavailableError(
-            'Timed out when trying to get node info from Kubernetes cluster. '
-            'Please check if the cluster is healthy and retry. To debug, run: '
-            'kubectl get nodes') from None
+    nodes = kubernetes.core_api(context).list_node(
+        _request_timeout=kubernetes.API_TIMEOUT).items
```

Comment:
Seems we already have the `request_timeout` set here. Is this not covering all the cases the `retry_on_error` can handle?

Reply:
No, this just sets the timeout for the specific request. K8s API server may return 503 immediately, in which case we need to back off and retry